docs/core-concepts/bot-sequences/end-criteria.mdx

```diff
@@ -4,6 +4,440 @@ sidebar_label: 'End Criteria'
 
 # End Criteria
 
-import Placeholder from '../../partials/_placeholder.mdx';
+End Criteria are the triggers that determine when a segment of your test sequence has completed successfully. End Criteria
+can be as simple as waiting for an entity to appear, or as complex as waiting for computer vision to detect a specific object.
 
-<Placeholder />
\ No newline at end of file
+## Structure of End Criteria
+
+End Criteria are defined in Bot Segments within the `endCriteria` field. A segment may contain multiple end criteria.
+
+```json
+// Segment structure
+{
+    "name": "Wait for the player to appear",
+    "description": "Waits for the player to appear on the screen",
+    "keyFrameCriteria": [
+        {
+            "type": ..., // The type of end criteria, as a string
+            "transient": ..., // If true, allows future criteria to be evaluated at the same time
+            "data": {
+                ... // The data for the end criteria, as a JSON object
+            }
+        }
+    ],
+    "botAction": {}
+}
+```
+
+An end criteria has a `type`, `data`, and `transient` field. The `type` indicates the type of end criteria to be used, and the `data` field contains the parameters for that end criteria. 
+If `transient` is `true`, this means that this criteria can match at any time during the evaluation of this bot segment and that passing result will persist even if it takes multiple more frames before other criteria in this segment are matched. If `transient` is `false`, this means that this criteria and other non-transient criteria must all be true at the same time (any transient criteria must also have matched already).
+
+For example, an end criteria to wait for an entity to appear may look like this:
+
+```json
+{
+    "name": "Perform an action once the start button appears",
+    "description": "Performs an action once the start button appears",
+    "endCriteria":[
+        {
+            "type": "NormalizedPath",
+            "transient": false,
+            "data": {
+                "path": "UI Canvas/Start Button",
+                "count": 1,
+                "countRule": "GreaterThanEqual"
+            }
+        }
+    ],
+    "botAction": {...}
+}
+```
+
+:::info
+There are additional fields that can be set on a bot action and many of these `inputData` fields, such as an apiVersion. You can see the full
+specification in the [source code](https://github.com/Regression-Games/RGUnityBots/blob/main/src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/KeyFrameCriteria.cs).
+:::
+
+## End Criteria Types
+
+There are many different types of end criteria that can be used in a bot segment, each useful in their own way.
+The list of supported types are as follows:
+
+* `NormalizedPath`: This end criteria waits for a specific game object to exist or not exist in the scene.
+* `PartialNormalizedPath`: Similar to `NormalizedPath`, but supports partial matches of the path string.
```

Comment:
We should note that `NormalizedPath` is far more performant than `PartialNormalizedPath` and should be preferred when possible.

Reply:
Added a note